test2.js

```
@@ -0,0 +1,40 @@
+if (!key in object) {
+    // operator precedence makes it equivalent to (!key) in object
+    // and type conversion makes it equivalent to (key ? "false" : "true") in object
+}
+
+/* eslint-disable no-alert, no-console */
+
+function badFunction() {
+  alert('This is an alert!'); 
+  console.log('Logging to the console.'); 
+  let x = 10;
+  if (x = 20) {
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)